lib/wda/utils.js

```diff
@@ -219,20 +220,33 @@ CODE_SIGN_IDENTITY = ${signingId}
 async function setXctestrunFile (deviceInfo, sdkVersion, bootstrapPath, wdaRemotePort) {
   const xctestrunFilePath = await getXctestrunFilePath(deviceInfo, sdkVersion, bootstrapPath);
   const xctestRunContent = await plist.parsePlistFile(xctestrunFilePath);
-  const updateWDAPort = {
-    WebDriverAgentRunner: {
-      EnvironmentVariables: {
-        USE_PORT: wdaRemotePort
-      }
-    }
-  };
-
+  const updateWDAPort = getWDAPort(deviceInfo.platformNameLowerCase, wdaRemotePort);
   const newXctestRunContent = _.merge(xctestRunContent, updateWDAPort);
   await plist.updatePlistFile(xctestrunFilePath, newXctestRunContent, true);
 
   return xctestrunFilePath;
 }
 
+/**
+ * Return the WDA object which has USER_PORT
+ * @param {DeviceInfo}
+ * @param {string} version - The Xcode SDK version of OS.
+ * @return {object} returns an runner object which has USE_PORT
+ */
+function getWDAPort (platformNameLowerCase, wdaRemotePort) {
+  const runner = platformNameLowerCase === 'tvos'
+    ? 'WebDriverAgentRunner_tvOS'
+    : 'WebDriverAgentRunner';
+
+  const wdaPort = {};
```

Comment:
```
const wdaPort = {
  runner: ...
}
```

Reply:
Would like to make there like:

```
{
    WebDriverAgentRunner: { // or WebDriverAgentRunner_tvOS: ...
        ...
    }
}
```
It looks the key will be `runner`.

Reply:
correction: use `[runner]` to make it dynamic